mitmproxy/proxy/mode_servers.py

```diff
@@ -307,18 +317,19 @@ async def listen(
                     )
                 except Exception:  # pragma: no cover
                     logger.debug("Failed to listen on '::', listening on IPv4 only.")
-                    return [ipv4]
+                    servers.append(ipv4)
```

Comment:
Nit: We can do this unconditionally right after the ipv4 assignment now, and then only .append ipv6 in the `else` clause. And/or integrate the else clause into the try part. :)